drivers/i2c/i2c_mcux_lpi2c.c

```diff
@@ -47,6 +58,9 @@ struct mcux_lpi2c_config {
 };
 
 struct mcux_lpi2c_data {
+#ifdef CONFIG_MMU
```

Comment:
is this (and the others) `ifdef` really useful? If `!DEVICE_MMIO_IS_IN_RAM` a lot (all?) of defines are already collapsing to NOP

Reply:
Not really understanding the point. What do you suggest to distinguish MPU and MCU?

Reply:
What I mean is `DEVICE_MMIO_NAMED_RAM` is already collapsing to nothing when `!defined(CONFIG_MMU)` (see in `include/zephyr/sys/device_mmio.h`) so this `ifdef` is redundant.

Reply:
Oh, so just remove it here.